packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -363,6 +411,113 @@ contract OPContractsManager is ISemver {
         return output;
     }
 
+    /// @notice addGameType deploys a new dispute game and links it to the DisputeGameFactory. The inputted _gameConfigs
+    /// must be added in ascending GameType order.
+    function addGameType(AddGameInput[] memory _gameConfigs) external returns (AddGameOutput[] memory) {
+        if (address(this) == address(thisOPCM)) revert OnlyDelegatecall();
+        if (_gameConfigs.length == 0) revert InvalidGameConfigs();
+
+        AddGameOutput[] memory outputs = new AddGameOutput[](_gameConfigs.length);
+        Blueprints memory bps = thisOPCM.blueprints();
+
+        // Store last game config to as an int64 so that we can ensure that the same game config is not added twice.
+        // An int64 is OK to use here since the GameType is a uint32. We start at -1 since 0 is a valid GameType.
+        int64 lastGameConfig = -1;
+
+        for (uint256 i = 0; i < _gameConfigs.length; i++) {
+            AddGameInput memory gameConfig = _gameConfigs[i];
+
+            // This conversion is safe because the GameType is a uint32, which will always fit in an int64.
+            int64 gameTypeInt = int64(uint64(gameConfig.disputeGameType.raw()));
+            // Ensure that the game configs are added in ascending order, and not duplicated.
+            if (lastGameConfig >= gameTypeInt) revert InvalidGameConfigs();
+            lastGameConfig = gameTypeInt;
+
+            // Grab the FDG from the SystemConfig.
+            IFaultDisputeGame fdg = IFaultDisputeGame(
+                address(
+                    IDisputeGameFactory(gameConfig.systemConfig.disputeGameFactory()).gameImpls(
+                        GameTypes.PERMISSIONED_CANNON
+                    )
+                )
+            );
```

Comment:
Comment and var name says we're getting FDG, but game type is using PDG. Chains are not guaranteed to have an FDG so this seems like it should be:
```suggestion
            // Grab the PDG from the SystemConfig.
            IPermissionedDisputeGame pdg = IPermissionedDisputeGame(
                address(
                    IDisputeGameFactory(gameConfig.systemConfig.disputeGameFactory()).gameImpls(
                        GameTypes.PERMISSIONED_CANNON
                    )
                )
            );
```

Reply:
This is how it was before, but I deliberately changed it. It's confusing either way:

- The PDG interface is a superset of the FDG. Therefore the interface that's guaranteed to be valid is FDG, not PDG.
- In the branch below I downcast to PDG in the `gameConfig.permissioned` branch.